Library/src/Systems/Assets/Material.cpp

```diff
@@ -0,0 +1,280 @@
+
+	// Base Color Factor
+	Serializer.SerializeMember(
+		"baseColorFactor",
+		csp::common::Array<float> {Obj.BaseColorFactor.X, Obj.BaseColorFactor.Y, Obj.BaseColorFactor.Z, Obj.BaseColorFactor.W});
+
+	// Metallic Factor
+	Serializer.SerializeMember("metallicFactor", Obj.MetallicFactor);
+
+	// Roughness Factor
+	Serializer.SerializeMember("roughnessFactor", Obj.RoughnessFactor);
+
+	// Emissive Factor
+	Serializer.SerializeMember("emissiveFactor", csp::common::Array<float> {Obj.EmissiveFactor.X, Obj.EmissiveFactor.Y, Obj.EmissiveFactor.Z});
+
+	// Textures
+	Serializer.SerializeMember("baseColorTexture", Obj.BaseColorTexture);
+	Serializer.SerializeMember("metallicRoughnessTexture", Obj.MetallicRoughnessTexture);
+	Serializer.SerializeMember("normalTexture", Obj.NormalTexture);
+	Serializer.SerializeMember("occlusionTexture", Obj.OcclusionTexture);
+	Serializer.SerializeMember("emissiveTexture", Obj.EmissiveTexture);
+}
+
+void FromJson(const csp::json::JsonDeserializer& Deserializer, csp::systems::GLTFMaterial& Obj)
+{
+	// Name
+	Deserializer.DeserializeMember("name", Obj.Name);
+
+	// Shader Type
+	uint32_t ShaderType;
+	Deserializer.DeserializeMember("shaderType", ShaderType);
+
+	Obj.Type = static_cast<csp::systems::EShaderType>(ShaderType);
+
+	// Version
+	Deserializer.DeserializeMember("version", Obj.Version);
+
+	// Alpha Mode
+	uint32_t AlphaMode;
+	Deserializer.DeserializeMember("alphaMode", AlphaMode);
+	Obj.AlphaMode = static_cast<csp::systems::EAlphaMode>(AlphaMode);
+
+	// Alpha Cutoff
+	Deserializer.DeserializeMember("alphaCutoff", Obj.AlphaCutoff);
+
+	// Double Sided
+	Deserializer.DeserializeMember("doubleSided", Obj.DoubleSided);
+
+	// Base Color Factor
+	csp::common::Array<float> BaseColorFactorArray;
+	Deserializer.DeserializeMember("baseColorFactor", BaseColorFactorArray);
+
+	Obj.BaseColorFactor = csp::common::Vector4(BaseColorFactorArray[0], BaseColorFactorArray[1], BaseColorFactorArray[2], BaseColorFactorArray[3]);
+
+	// Metallic Factor
+	Deserializer.DeserializeMember("metallicFactor", Obj.MetallicFactor);
+
+	// Roughness Factor
+	Deserializer.DeserializeMember("roughnessFactor", Obj.RoughnessFactor);
+
+	// Emissive Factor
+	csp::common::Array<float> EmissiveFactorArray;
+	Deserializer.DeserializeMember("emissiveFactor", EmissiveFactorArray);
+
+	Obj.EmissiveFactor = csp::common::Vector3(EmissiveFactorArray[0], EmissiveFactorArray[1], EmissiveFactorArray[2]);
+
+	// Textures
+	Deserializer.DeserializeMember("baseColorTexture", Obj.BaseColorTexture);
+	Deserializer.DeserializeMember("metallicRoughnessTexture", Obj.MetallicRoughnessTexture);
+	Deserializer.DeserializeMember("normalTexture", Obj.NormalTexture);
+	Deserializer.DeserializeMember("occlusionTexture", Obj.OcclusionTexture);
+	Deserializer.DeserializeMember("emissiveTexture", Obj.EmissiveTexture);
+}
+
+namespace csp::systems
+{
+
+GLTFMaterial::GLTFMaterial(const csp::common::String& InAssetCollectionId, const csp::common::String& InAssetId)
```

Comment:
I think this should either inherit from a base class or have a common interfact for the base "Material". I'd argue gltf material should be in its own file.
This is because future shader types may share almost 0 properties with gltf shader (other than name, type etc)

Reply:
I assume Material base should have:

- Name
- Type 
- AssetCollectionId
- AssetId